frontend/src/metabase/query_builder/actions/core/initializeQB.ts

```diff
@@ -247,17 +247,17 @@ async function handleQBInit(
     return;
   }
 
-  if (hasCard) {
-    await handleDashboardParameters(card, {
+  if (deserializedCard?.dashcardId) {
+    card = await propagateDashboardParameters({
+      card,
       deserializedCard,
       originalCard,
       dispatch,
-      getState,
     });
-  } else {
-    if (options.metric) {
-      uiControls.isShowingSummarySidebar = true;
-    }
+  }
+
+  if (!hasCard && options.metric) {
+    uiControls.isShowingSummarySidebar = true;
```

Comment:
I shuffled the `hasCard` check down to here. I believe these are orthogonal concerns anyway.